tools/dsom_token_auditor.py

```
@@ -0,0 +1,58 @@
+import tiktoken
+
+def count_tokens(text: str, model: str = "gpt-4") -> int:
+    """Counts tokens using tiktoken (a proxy for modern LLMs)."""
+    try:
+        encoding = tiktoken.encoding_for_model(model)
+        return len(encoding.encode(text))
+    except Exception as e:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)